File: leetcode/java/TwoSum.java
package com.programming;
import java.util.*;

public class TwoSum {

        private static int[] twoSum(int[] nums, int target) {
            Map<Integer, Integer> numMap = new HashMap<>();
            for (int i = 0; i < nums.length; i++) {
                int complement = target - nums[i];
                if (numMap.containsKey(complement)) {
                    return new int[] { numMap.get(complement), i };
                } else {
                    numMap.put(nums[i], i);
                }
            }
            return new int[] {};
        }


    public static void main(String[] args) {
        int ans[] = twoSum(new int[]{1,5,4}, 5);
        Arrays.stream(ans).forEach(System.out::println);
    }

}

File: leetcode/js/two-sum.js
function twoSum(a,b){
  var x = {}
  for(let i=0; i<a.length;i++){
    let diff = b - a[i] 
     if(x[diff] !== undefined){
         return [x[diff],i]
      }
     x[a[i]]=i
  }
  return
}

twoSum([1,3,2],3)